modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -211,14 +212,26 @@ trait MutationMapping[F[_]] extends Predicates[F] {
   private lazy val CloneObservation: MutationField =
     MutationField("cloneObservation", CloneObservationInput.Binding) { (input, child) =>
       services.useTransactionally {
-        observationService.cloneObservation(input).map { r =>
+
+        val clone: ResultT[F, (Program.Id, Observation.Id)] = 
+          ResultT(observationService.cloneObservation(input))
+
+        // this will do nothing if input.asterism is Absent
+        def setAsterism(pid: Program.Id, oid: Observation.Id): ResultT[F, Unit] =
```

Comment:
Just curious as to why you are doing this here, instead of doing it in `ObservationService.cloneObservation` instead of calling `asterismService.setAsterism` there. It seems you potentially create the asterism (via cloning), then replace it.

Reply:
Oh, sorry. Didn't notice it was still a draft.

Reply:
This is the way it's done in `updateObservations` so I decided to do it the same way here, but yes it seems like this operation should be pushed down so that the mutations themselves only do a single service call. I'll create an issue to revisit this.